### ISSUE TITLE
Which Node version is best to use for Flok?

### ISSUE THREAD

**Issue by tmhglnd** [munshkr/flok#170]
Hi! 

I'm wondering what is now the best version of Node to use with Flok? Previously I believe `v12` was the best also for combination with Hydra. But running `yarn dev` for the local `'flok-web` package in v12 results in the following error: 
```
/Users/timohoogland/Drive/work/code/js/flok/packages/web/lib/Server.js:121
          nextApp.hotReloader?.onHMR(req, socket, head);
                              ^

SyntaxError: Unexpected token '.'
```

Using version 14 and 18 seems to work with `flok-web` when running `yarn dev`, but running `yarn start` fails silently (or at least the server is not started) and I only see the following printed:
```
yarn run v1.22.19
$ cross-env NODE_ENV=production node index.js
✨  Done in 0.50s.
```

When running `flok-repl` however I can't use version 18. This results in the following error: 
```
unable connect to the server Error: connect ECONNREFUSED ::1:3000
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16) {
  errno: -61,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '::1',
  port: 3000
}
Connection is closed
```
Changing node to run in v14 works with the `repl` (I did not try other versions).

So my conclusion currently: everything works in v14, and for `flok-web` I have to use `yarn dev` instead of `yarn start`. Is this expected or am I missing something? 

Best,
Timo

**Comment by munshkr** [comment]
I think I [just fixed](https://github.com/munshkr/flok/commit/9a53e62ea8855dcac3a552295ec8f6a48606a798) the `yarn start` not doing anything problem, can you try again? (also @yaxu might be interested)

It should work with latest Node versions, that is, LTS v18 and probably v19 too, though I haven't tried with it.

**Comment by tmhglnd** [comment]
Hey, thanks for the quick reply! I don't know, maybe I'm doing something wrong, but in any case just pulled the latest but for me it didn't change anything. 

`yarn start` just prints: ```yarn run v1.22.19
$ cross-env NODE_ENV=production node server.js
✨  Done in 0.29s.```
both in 18 LTS and 14

`flok-repl` still doesnt work in 18

**Comment by munshkr** [comment]
That's weird.. I'm using Node 18.15 and `web` works for me... What's the exact version of Node you're using? 

The connection refused error on `repl` is because it's trying to connect to the websockets server that `web` starts but can't, probably because web didn't start correctly in the first place.

**Comment by munshkr** [comment]
Ah, I found the problem. If the server fails to start, it ends silently. Maybe you have another process ocupying the 3000 port somewhere? I'll try to fix that in the mean time.

**Comment by yaxu** [comment]
Yes I'm seeing the same problems in node 18, `yarn run dev` working, others exiting silently

**Comment by munshkr** [comment]
Can you try again? It should throw an error now...

**Comment by tmhglnd** [comment]
Yes, now it throws an error, which is helpful, and after running `yarn build` I can run `yarn start` and it works! In v18.15.0.

So only the `flok-repl` still gets an error in v18.15.0, even tho the website is started properly and I created a session. It works if I change to v14
